theano/tensor/nnet/nnet.py

```diff
@@ -2014,7 +2014,8 @@ def binary_crossentropy(output, target):
     TODO : Rewrite as a scalar, and then broadcast to tensor.
 
     """
-    return -(target * tensor.log(output) + (1.0 - target) * tensor.log(1.0 - output))
+    one = np.asarray(1.0).astype(theano.config.floatX)
```

Comment:
This is what should happen automatically. So I don't see why this is needed? Maybe if floatX is float64 and target and output are float32, there is a problem here. I'm not sure about that. But if output, target and floatX are float32, I don't see a need for this fix.

What is your exact case? dtypes of target and output?